Add Python 3 linting to the Makefile

noahl opened the pull request:
This will let us catch Python 3 errors before the code gets to
Travis. This commit adds Python 3 linting, and also updates Rho to
pass Python 3 linting cleanly.

Commit:
Add Python 3 linting to the Makefile

This will let us catch Python 3 errors before the code gets to
Travis. This commit adds Python 3 linting, and also updates Rho to
pass Python 3 linting cleanly.

elyezer commented on rho/translation.py:
```diff
@@ -19,6 +19,7 @@
 def get_translation():
     """Obtains the locale based translation setup for rho"""
     if hasattr(T, 'ugettext'):
+        # pylint: disable=no-member
```
This disable could be avoided by writing something like this:

```python
return getattr(T, 'ugettext', T.gettext)
```

PS: also it would simplify the function body.

noahl replied:
That would work, but I would find it less clear to use the `getattr` method than the if statement.

noahl replied:
What do you think?

elyezer commented on rho/translation.py:
```diff
@@ -19,6 +19,7 @@
 def get_translation():
```
not sure if this is still used because I see on other modules this import `from ansible.module_utils.translation import _`, maybe this module can be removed on a future iteration.

Commit:
More Python 3 fixes

Commit:
Update .travis.yml for Python 3 changes

Commit:
Remove dependence on pylint-3 name.

Use 'python3 -m pylint' instead.

Commit:
Separate Python 3 commands in Makefile

This resolves problems on Travis, which doesn't use the -3 prefix for
Python 3 executables.

chambridge reviewed (approved):
LGTM.
